### PR TITLE
204 monitoring infrastructure

### LINKED ISSUE #204: Monitoring infrastructure for mirror nodes
**Problem**
We need a monitoring infrastructure for the mirror nodes.
**Solution**
It should detect when anything fails in the pipeline from the files in S3 collector to the parser to DB to the REST APIs.

**Alternatives**
Manual monitoring

### PR DESCRIPTION
**Pull request by atul-hedera**
**Detailed description**:
To monitor a live deployment of a Hedera mirror node, this code consists of monitoring APIs and an optional monitoring dashboard. Essentially, this code runs on an external server, and periodically pings a live mirror node deployment to check the results. These are expected to be integrated with the devops alerting infrastructure. An optional dashboard can be used to see the red/green status.

This code does not share anything with the rest-api code branch.
 
**Which issue(s) this PR fixes**:
Fixes #204 

**Special notes for your reviewer**:
Ran a live 'install and run' demo with the devops team.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: rest-api/monitoring/monitor_apis/accounts.monitor.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    const response = await fetch(url);
+	const data = await response.json();
+	
+	common.logResult (server, url, 'getAccountsNoParams',
+		(data.accounts.length === 1000) ?
+		{result: true, msg: 'Received 1000 accounts'} : 
+		{result: false, msg: 'Received less than 1000 accounts'});
+
+	const txSec = data.accounts[0].balance.timestamp.split('.')[0];
+	const currSec = Math.floor(new Date().getTime() / 1000);
```
Comment (mike-burrage-hedera) [inline rest-api/monitoring/monitor_apis/accounts.monitor.js]: Make the "1000" configurable with a default to 1000. I know we have the hard-coded default limit on the rest-api side at 1000, but at least putting the "1000" in one place makes modification easier should we change that default on the rest-api side. If not configurable, then at least a constant in 1 place in the code. Other checks (balances/transactions) should use the same value.
Check for places where this is logged/printed out as well.

File: rest-api/monitoring/monitor_apis/accounts.monitor.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+/**
+ * Executes the /accounts API with no parameters
+ * Expects the response to have 1000 entries, and timestamp in 
+ * the FRESHNESS_EXPECTATION minutes
+ * @param {Object} server The server to run the test against
+ * @return {} None. updates testAccounts variable
+ */
+const getAccountsNoParams = async (server) => {
+    const FRESHNESS_EXPECTATION = 20; // minutes
```
Comment (mike-burrage-hedera) [inline rest-api/monitoring/monitor_apis/accounts.monitor.js]: Global configuration option, please. Same for balances.

File: rest-api/monitoring/monitor_apis/transactions.monitor.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+/**
+ * Executes the /transactions API with no parameters
+ * Expects the response to have 1000 entries, and timestamp in 
+ * the FRESHNESS_EXPECTATION minutes
+ * @param {Object} server The server to run the test against
+ * @return {} None. updates testTransactions variable
+ */
+const getTransactionsNoParams = async (server) => {
+    const FRESHNESS_EXPECTATION = 5; // minutes
```
Comment (mike-burrage-hedera) [inline rest-api/monitoring/monitor_apis/transactions.monitor.js]: Global constant or configurable parameter please.

File: rest-api/monitoring/monitor_apis/accounts.monitor.js (hunk 1 above)
Comment (atul-hedera) [reply rest-api/monitoring/monitor_apis/accounts.monitor.js]: Javascript getTime() returns milliseconds, and this is dividing by 1000 to get to seconds. Will add a comment to make it clear.

File: rest-api/monitoring/monitor_apis/accounts.monitor.js (hunk 2 above)
Comment (atul-hedera) [reply rest-api/monitoring/monitor_apis/accounts.monitor.js]: OK. Will take care of it.

File: rest-api/monitoring/monitor_apis/transactions.monitor.js (hunk 3 above)
Comment (atul-hedera) [reply rest-api/monitoring/monitor_apis/transactions.monitor.js]: OK. Will take care of it.

File: rest-api/monitoring/monitor_apis/accounts.monitor.js (hunk 1 above)
Comment (atul-hedera) [reply rest-api/monitoring/monitor_apis/accounts.monitor.js]: Done

File: rest-api/monitoring/monitor_apis/accounts.monitor.js (hunk 2 above)
Comment (atul-hedera) [reply rest-api/monitoring/monitor_apis/accounts.monitor.js]: Done

File: rest-api/monitoring/monitor_apis/transactions.monitor.js (hunk 3 above)
Comment (atul-hedera) [reply rest-api/monitoring/monitor_apis/transactions.monitor.js]: Done

**Pull request merged**